src/promptflow-tools/promptflow/tools/azure_content_safety.py

```diff
@@ -0,0 +1,319 @@
+            content: str,
+            blocklists: List[str] = [],
+    ) -> dict:
+        url = self.build_url(media_type)
+        headers = self.build_headers()
+        request_body = self.build_request_body(media_type, content, blocklists)
+        payload = json.dumps(request_body)
+        response = requests.post(url, headers=headers, data=payload)
+        print("status code: " + response.status_code.__str__())
+        print("response txt: " + response.text)
+        res_content = response.json()
+        if response.status_code != 200:
+            raise DetectionError(res_content["error"]["code"], res_content["error"]["message"])
+        return res_content
+
+    def get_detect_result_by_category(self, category: Category, detect_result: dict) -> Union[int, None]:
+        if category == Category.Hate:
+            return detect_result.get("hateResult", None)
+        elif category == Category.SelfHarm:
+            return detect_result.get("selfHarmResult", None)
+        elif category == Category.Sexual:
+            return detect_result.get("sexualResult", None)
+        elif category == Category.Violence:
+            return detect_result.get("violenceResult", None)
+        else:
+            raise ValueError(f"Invalid Category {category}")
+
+    def get_detect_result_by_category_1001(self, category: Category, detect_result: dict) -> Union[int, None]:
+        category_res = detect_result.get("categoriesAnalysis", None)
+        for res in category_res:
+            if category.name == res.get("category", None):
+                return res
+        raise ValueError(f"Invalid Category {category}")
+
+    def make_decision(
+            self,
+            detection_result: dict,
+            reject_thresholds: Dict[Category, int],
+    ) -> Decision:
+        action_result = {}
+        final_action = Action.Accept
+        for category, threshold in reject_thresholds.items():
+            if threshold not in (-1, 0, 2, 4, 6):
+                raise ValueError("RejectThreshold can only be in (-1, 0, 2, 4, 6)")
+
+            cate_detect_res = self.get_detect_result_by_category(category, detection_result)
+            if cate_detect_res is None or "severity" not in cate_detect_res:
+                raise ValueError(f"Can not find detection result for {category}")
+
+            severity = cate_detect_res["severity"]
+            action = Action.Reject if threshold != -1 and severity >= threshold else Action.Accept
+            action_result[category] = action
+            if action.value > final_action.value:
+                final_action = action
+
+        if (
+                "blocklistsMatchResults" in detection_result
+                and detection_result["blocklistsMatchResults"]
+                and len(detection_result["blocklistsMatchResults"]) > 0
+        ):
+            final_action = Action.Reject
+
+        print(final_action.name)
+        print(action_result)
+        return Decision(final_action, action_result)
```

Comment:
create specific exception for error handing, please refine all exception in your tool